FlowCrypt/Controllers/Inbox/InboxViewController+TableView.swift

```diff
@@ -341,55 +360,103 @@ extension InboxViewController {
         return inboxInput[safe: index]
     }
 
-    func perform(action: MessageAction, at indexPath: IndexPath) {
-        guard let inboxItem = inboxItem(at: indexPath) else { return }
+    private func performPermanentDeleteAction(on inboxItems: [InboxItem]) {
+        showPermanentDeleteThreadAlert(onAction: { [weak self] _ in
+            guard let self else { return }
+            for inboxItem in inboxItems {
+                self.handleOperation(inboxItem: inboxItem, action: .permanentlyDelete)
+                Task {
+                    try? await self.threadOperationsApiClient.delete(id: inboxItem.threadId)
+                }
+            }
+            updateNavigationBar()
+        })
+    }
 
-        Task {
-            do {
-                try await messageActionsHelper.perform(
+    private func performOtherAction(on inboxItems: [InboxItem], action: MessageAction) {
```

Comment:
let's name it `performMessageAction(_ action: MessageAction, inboxItems: [InboxItem]`